src/base/TilesRendererBase.d.ts

```diff
@@ -20,9 +40,9 @@ export class TilesRendererBase {
 
 	constructor( url?: string );
 	update() : void;
-	registerPlugin( plugin: object ) : void;
-	unregisterPlugin( plugin: object | string ) : boolean;
-	getPluginByName( plugin: object | string ) : object;
+	registerPlugin( plugin: TPlugin ) : void;
+	unregisterPlugin( plugin: TPlugin | string ) : boolean;
+	getPluginByName( plugin: string ) : TPlugin;
```

Comment:
Will this templating not prevent "base" plugins from being registered on `TilesRenderer`?

Reply:
No, because of two reasons:
- TypeScript uses structural typing and not nominal typing, so the "name" of the type of the plugin does not matter, only the "shape", and...
- since all methods in the plugin are optional, structural typing will not care that a "base" plugin is "missing" methods from the derived plugin type.